README.md
# i2b2 Tools
A collection of tools created by [Washington University in St. Louis Pediatrics Development Team](http://wus.tl/pcf) for use with [i2b2](http://i2b2.org)

If you are interested in contributing to the cause please feel free to contact us : appdev ( at ) kids.wustl.edu

Linux Shrine Installer/Changes.md
1. vm-install.sh
	* Added support for sqlserver parameter as there was no way to detect the use of MSSQL like postgres or oracle. This could probably be added to common.rc as a setup parameter instead.

2. i2b2/sqlserver/clean*.*
	* I have yet to update the clean position of the script so no changes to any of the clean* files under the i2b2/sqlserver folder

3. i2b2/sqlserver/configure_hive.sh
	* added additional required variables to support the MSSQL syntax in the interpolated files
	* changed to use the java tsql class

4. i2b2/sqlserver/configure_hive.sh
	* changed to use the java tsql class

5. i2b2/sqlserver/i2b2.rc
	* there are many changes in this file, mainly adding additional variables to support MSSQL configuration. variables were reused within the file where possible to minimize the number of changes needed to get the script to run against the default MSSQL configuration for i2b2. a few sections had to be added to support updating all of the i2b2 datasource files to point to MSSQL, lines 134-138 for example.

6. i2b2/sqlserver/ontology.sh
	* the main changes here are updating the variables that are used during interpolation and using java for execution

7. i2b2/sqlserver/prepare.sh
	* added a line to compile the tsql java class and a call to reconfigure-i2b2.sh

8. i2b2/sqlserver/reconfigure-i2b2.sh
	* replaces all of the *-ds.xml files in the jboss deployment folder with copies that point to MSSQL using the same process as the ont-ds.xml replacement in the configre_hive.sh file
	* also replaces a few values in the i2b2 properties files to update the schema values and database type

9. i2b2/sqlserver/skel/
	* the required changes to the sql files were minimal, mainly added the I2B2_DB_SCHEMA token and changed a few minor syntax issues
	* added “tokenized” copies of the remaining *-ds.xml files to support updating i2b2 to point to MSSQL

10. shring/sqlserver/adapter.sql, hub.sql, create_broadcaster_audit_table.sql
	* had to provide local copies of these files that have been modified to run on MSSQL

11. shrine/sqlserver/create_database.sql, create_user.sql
	* the mysql.sql file had to be split into these two files due to a limitation in adbc. someone with a bit more experience with jdbc may be able to figure out how to merge them into a mssql.sql file but this was the easiest way forward. basically the create database command must complete before the create user command can be run, jdbc does not support the GO batch delimiter to execute commands in batches so this required two separate calls to jdbc.

12. shrine/sqlserver/install.sh
	* changed to call mssql.sh instead of mysql.sh
	* also changed the path to common.rc by adding another ../ since it is now in a subfolder

13. shrine/sqlserver/install_prereqs.sh
	* added a call to install svn… noticed this was not installed because i was coping the quick_install folder from my dev box to the vm instead of using svn to download it. figured i would add it here for good measure.
	* added a line to compile the java tsql class

14. shrine/sqlserver/mssql.sh
	* changed to use different variables
	* changed to use jdbc
	* no longer downloads the sql files as mentioned in #10

15. shrine/sqlserver/shrine.rc
	* changed SHRINE_MYSQL_* to SHRINE_DB_*
	* added SHRINE_DB_ADMIN_* to support logging into MSSQL as a server admin to create the new database
	* added SHRINE_DB_SCHEMA to support MSSQL schema

16. shrine/sqlserver/tomcat.sh
	* changed the interpolation of shrine.xml to use the new variables

17. shrine/sqlserver/skel/shrine.conf
	* changed the databaseType to mssql

18. shrine/sqlserver/skel/shrine.xml
	* changed the jdbc connection string



SQL Side:
                update SQL PM Cell Data to localhost


Web Side:
                sqljdbc.jar - copy from JBoss to Tomcat/lib
                install tomcat to c:\opt\shrine\tomcat (no spacing)
                server.xml:
                                Uncomment SSL/TLS Connector
                                                port="6443"
                                                protocol="HTTP/1.1"
                                                Add keystoreFile and keystorePass to 6443 Connector (able to use relative path/does not assume C:)
                                Edit 6060 redirectPort="6443"
                shrine.conf:
                                pmEndpoint url should be localhost:9090
                                ontEndpoint url should be localhost:9090
                                crcEndpoint url should be localhost:9090
                                shrineDatabaseType should be "sqlserver"
                              (M)  keystore file should be edited to match windows directory and escape chars (\\) added


I2B2 Windows Installer/common.ps1
Write-Host "importing configuration values"



$_SHRINE_VERSION = "1.18.2"

$_I2B2_DOMAIN_ID = "i2b2demo"

$_SHRINE_IP = "localhost"
$_SHRINE_MSSQL_SERVER = "localhost"

$_NEXUS_URL_BASE = "http://repo.open.med.harvard.edu/nexus/content/groups/public/net/shrine"
$_SHRINE_SVN_URL_BASE = "https://open.med.harvard.edu/svn/shrine/releases/$_SHRINE_VERSION"

$_I2B2_IP_DEFAULT = "localhost:9090"

$_I2B2_PM_IP = $_I2B2_IP_DEFAULT
$_I2B2_ONT_IP = $_I2B2_IP_DEFAULT
$_I2B2_CRC_IP = $_I2B2_IP_DEFAULT

$_SHRINE_PORT = "6060"
$_SHRINE_SSL_PORT = "6443"



#Shrine, including the Tomcat application server that Shrine runs in, will be installed here.
$_SHRINE_HOME = "C:\opt\shrine"

#The directory containing the Tomcat application server that Shrine runs in.
$_SHRINE_TOMCAT_HOME = "$_SHRINE_HOME\tomcat"

#A directory on Tomcat'sclasspath, from which shrine.conf will be loaded 
$_SHRINE_TOMCAT_LIB = "$_SHRINE_TOMCAT_HOME\lib"

#The primary Shrine config file
$_SHRINE_CONF_FILE = "$_SHRINE_TOMCAT_LIB\shrine.conf"

#Tomcat's main configuration file.
$_SHRINE_TOMCAT_SERVER_CONF = "$_SHRINE_TOMCAT_HOME\conf\server.xml"

#The location of the Shrine web app's context configuration file.
$_SHRINE_TOMCAT_APP_CONF = "$_SHRINE_TOMCAT_HOME\conf\Catalina\localhost\shrine.xml"

#The human-readable name of the Shrine node being installed.  This string will appear in the web UI as a label on a source of query results.
$_SHRINE_NODE_NAME = "Harvard Affiliated Teaching Hospital"

#The cryptographic keystore file used by Shrine.
$_KEYSTORE_FILE = "$_SHRINE_HOME\shrine.keystore"

#Default password for the keystore
$_KEYSTORE_PASSWORD = "changeit"

#Human-readable name for the cryptographic certificate generated for this Shrine node.
$_KEYSTORE_ALIAS = $_SHRINE_IP

#Human-readable name for the cryptographic certificate generated for this Shrine node.
$_KEYSTORE_HUMAN = $_SHRINE_NODE_NAME

#City where the node resides; will be included in generated cryptographic certificate.
$_KEYSTORE_CITY = "Boston"

#State where the node resides; will be included in generated cryptographic certificate.
$_KEYSTORE_STATE = "MA"

#Country where the node resides; will be included in generated cryptographic certificate.
$_KEYSTORE_COUNTRY = "US"

#The host at which MySQL is accessible.  MySQL is needed by Shrine for logging purposes.
$_SHRINE_MYSQL_HOST = "localhost"

#The MySQL user Shrine should log in as.
$_SHRINE_MYSQL_USER = "shrine"

#The password for the MySQL user that Shrine should log in as.
$_SHRINE_MYSQL_PASSWORD = "demouser"

#The database name for the MSSQL server instance
$_SHRINE_MSSQL_DB = "shrine"

#The MSSQL user Shrine should log in as.
$_SHRINE_MSSQL_USER = "shrine_installer"

#The password for the MSSSQL user that Shrine should log in as.
$_SHRINE_MSSQL_PASSWORD = "demouser"

#The i2b2 user Shrine should use when making queries to an i2b2 hive.
$_SHRINE_ADAPTER_I2B2_USER = "demo"

#The password for the i2b2 user Shrine should use when making queries to an i2b2 hive.
$_SHRINE_ADAPTER_I2B2_PASSWORD = "demouser"

#The i2b2 project Shrine should use when making queries to an i2b2 hive.
$_SHRINE_ADAPTER_I2B2_PROJECT = "Demo"

#The i2b2 domain Shrine should use when making queries to an i2b2 hive.
#NB: This needs to match I2B2_DOMAIN_ID:(
$_SHRINE_ADAPTER_I2B2_DOMAIN = "i2b2demo"
